specification/servicefabric/resource-manager/Microsoft.ServiceFabric/2016-09-01/servicefabric.json

```diff
@@ -356,6 +395,96 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.ServiceFabric/locations/{location}/environments/{environment}/clusterVersions/{clusterVersion}": {
+      "get": {
+        "tags": [
+          "ClusterVersion"
+        ],
+        "operationId": "ClusterVersions_GetByEnvironment",
+        "description": "Get cluster code versions by environment",
+        "x-ms-examples": {
+          "Get cluster version": {
+            "$ref": "./examples/ClusterVersionsGetByEnvironment_example.json"
+          }
+        },
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/locationForClusterCodeVersions"
+          },
+          {
+            "$ref": "#/parameters/environment"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          },
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/clusterVersion"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK - Get cluster code versions successfully",
+            "schema": {
+              "$ref": "#/definitions/ClusterCodeVersionsListResult"
```

Comment:
Do you have an example response? It sounds to me that multiple resources are sharing the same id here...

Pardon me for being a newbie to your service, but I'm just trying to understand from an average developer's perspective. Developers using `GetXXX()` methods are used to getting a single response instead of trying to page through a list.